frontend/src/pages/projects/screens/spawner/dataConnection/ExistingDataConnectionField.tsx

```diff
@@ -28,6 +95,59 @@ const ExistingDataConnectionField: React.FC<ExistingDataConnectionFieldProps> =
     );
   }
 
+  const onToggleClick = () => {
+    setOpen(!isOpen);
+  };
+
+  const onTextInputChange = (_event: React.FormEvent<HTMLInputElement>, value: string) => {
+    setInputValue(value);
+    setFilterValue(value);
+  };
+
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle
+      ref={toggleRef}
+      variant="typeahead"
+      aria-label="Options menu"
+      onClick={onToggleClick}
+      isDisabled={empty}
+      isExpanded={isOpen}
+      isFullWidth
+    >
+      <TextInputGroup isPlain>
+        <TextInputGroupMain
+          value={inputValue}
+          onClick={onToggleClick}
+          onChange={onTextInputChange}
+          id="typeahead-select-input"
+          autoComplete="off"
+          innerRef={textInputRef}
+          placeholder={placeholderText}
+          role="combobox"
+          isExpanded={isOpen}
+          aria-controls="select-typeahead-listbox"
```

Comment:
A reusable typeahead component should be made that can remove duplicate code every time we need to use this component. 

There's some issues with using the current typeahead. After typing in a value, there's no way for me to auto complete using the keyboard like I could with the old implementation.

Once I've typed a matching value, there's no way to navigate away from the input field using keyboard.